### PR TITLE
added unit tests for utils/file/file-picker

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Increase coverage from 0% to 95.16%

image

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-59561

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.
- [ ] Have run E2E tests by adding label `E2E iOS tests for PR`.

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by saturninoabril — APPROVED** [review]
🙌

File: app/utils/file/file_picker/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,803 @@
...
+    beforeAll(() => {
+        Platform.select = ({android, ios, default: dft}: any) => {
+            if (Platform.OS === 'android' && android) {
+                return android;
+            } else if (Platform.OS === 'ios' && ios) {
+                return ios;
+            }
+
+            return dft;
+        };
```
Comment (rahimrahman) [inline app/utils/file/file_picker/index.test.ts:63]: Since this is repeated a few times, perhaps extract this into a higher scope (1 describe() higher)?

**Review by rahimrahman — COMMENTED** [review]
The test file is long, and extensive, because fileUtilPicker is big (375 lines of code) and does a lot of things.  Amazing job to get to 95+%.

I've noticed that certain parts of the test file were broken into it's own section.  Maybe some of the section(s) can go further with subsection?

Ex:

```
describe('FilePickerUtil', () => {
    describe('getting appropriate permission messages via getPermissionMessages()', () => {
        test('should return correct permission messages for camera', () => {}
        test('should return correct permission messages for storage', () => {}
        test('should return correct permission messages for photo_ios', () => {}
    }
}
```

I feel like it may help with readability.

File: app/utils/file/file_picker/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,803 @@
...
+}));
+
+jest.mock('@utils/log', () => ({
+    logWarning: jest.fn(),
+}));
+
+jest.mock('@mattermost/rnutils', () => ({
+    getRealFilePath: jest.fn(),
+    isRunningInSplitView: jest.fn().mockReturnValue({isSplit: false, isTablet: false}),
+}));
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/utils/file/file_picker/index.test.ts:45]: doing that will cause issues with the testing of the helpers.. so leaving it like this

**Review by rahimrahman — APPROVED** [review]

**Pull request merged**